pkg/knuu/instance.go

```diff
@@ -538,6 +550,42 @@ func (i *Instance) SetServiceAccount(serviceAccount string) error {
 	return nil
 }
 
+// SetLivenessProbe sets the liveness probe of the instance
+// A live probe is a probe that is used to determine if the instance is still alive, and should be restarted if not
+// This function can only be called in the states 'Preparing' and 'Committed'
+func (i *Instance) SetLivenessProbe(livenessProbe *k8s.Probe) error {
+	if !i.IsInState(Preparing, Committed) {
+		return fmt.Errorf("setting liveness probe is only allowed in state 'Preparing' or 'Committed'. Current state is '%s'", i.state.String())
+	}
```

Comment:
[non-blocking]: this can be a good first issue to create a private func that does the checking for errors based on inputs of errors

Reply:
Do you mean something like this: https://github.com/celestiaorg/knuu/issues/15?

Reply:
Not really. You can just encapsulate the checker if into a single private func that will be called by 3 Public funcs providing it (private) with strings each